c/charsets.c

```diff
@@ -116,6 +116,33 @@ int convertCharset(char *input,
 
 #elif defined(__ZOWE_OS_ZOS)
 
+static const union {
+  char table[256];
+  uint64_t alignment;
```

Comment:
You can align storage using the [aligned variable attribute ](https://www.ibm.com/support/knowledgecenter/en/SSLTBW_2.3.0/com.ibm.zos.v2r3.cbclx01/var_attrib_aligned.htm). It annoyingly emits a warning message which I suppress. It would be interesting to measure if there is any performance difference between 4 byte, 8 byte and 16 byte alignment in 64-bit code. 

I have anecdotal evidence that a double world aligned `ex` instruction was much quicker but that was a long time ago and is probably obsolete.

Reply:
I did use it initially, then spent lots of time trying to suppress that message (turns out you can't use pragmas to do that) and decided to just use a union.

Reply:
Rather than duplicate the table I would have suggested you align the existing tables in xlate_tables.c.  Now the same table exists twice.

FYI, I usually use a union, like you did, to force alignment of storage.

Reply:
@gadyck , I agree, that would be a better approach. I'm thinking, if possible, how we could do that without having to change any existing code that uses these tables.